testbed/core/tests/test_api.py

```diff
@@ -65,4 +69,283 @@ def test_actor_not_found():
+        assert lola_count >= public_count
+        
+        # Both should have proper outbox ID
+        expected_outbox_id = build_outbox_id(actor.id)
+        assert public_data["id"] == expected_outbox_id
+        assert lola_data["id"] == expected_outbox_id
+    
+    # Test that demonstrates clear differences between public and LOLA responses
+    @pytest.mark.django_db
+    def test_side_by_side_authentication_comparison(self, lola_token):
+        actor = create_isolated_actor("comparison_test")
+        
+        # Public request
+        public_client = APIClient()
+        public_response = public_client.get(reverse("actor-detail", kwargs={"pk": actor.id}))
+        public_data = public_response.data
+        
+        # LOLA request
+        lola_client = APIClient()
+        lola_client.credentials(HTTP_AUTHORIZATION=f'Bearer {lola_token.token}')
+        lola_response = lola_client.get(reverse("actor-detail", kwargs={"pk": actor.id}))
+        lola_data = lola_response.data
+        
+        # Both should succeed
+        assert public_response.status_code == status.HTTP_200_OK
+        assert lola_response.status_code == status.HTTP_200_OK
+        
+        # Both should have identical basic fields
+        basic_fields = ["@context", "type", "id", "preferredUsername", "name", "previously"]
+        for field in basic_fields:
+            assert public_data[field] == lola_data[field]
+        
+        # Only LOLA should have enhanced fields
+        lola_fields = ["accountPortabilityOauth", "content", "blocked", "migration"]
+        for field in lola_fields:
+            assert field not in public_data
+            assert field in lola_data
+            assert isinstance(lola_data[field], str)  # Should be URL strings
+    
+    # Test that invalid tokens gracefully degrade to unauthenticated behavior
+    @pytest.mark.django_db
+    def test_invalid_token_graceful_degradation(self):
+        actor = create_isolated_actor("invalid_token_test")
+        client = APIClient()
+        
+        # Use completely invalid token
+        client.credentials(HTTP_AUTHORIZATION='Bearer invalid-nonexistent-token-xyz')
+        response = client.get(reverse("actor-detail", kwargs={"pk": actor.id}))
+        
+        # Should succeed with public data (graceful degradation)
+        assert response.status_code == status.HTTP_200_OK
+        
+        data = response.data
+        assert data["type"] == "Person"
+        # Should NOT have LOLA fields (invalid token treated as unauthenticated)
+        assert "accountPortabilityOauth" not in data
+        assert "content" not in data
+        assert "blocked" not in data
+    
+    # Test graceful handling of malformed authorization headers
+    @pytest.mark.django_db
+    def test_malformed_authorization_header_handling(self):
+        actor = create_isolated_actor("malformed_header_test")
+        client = APIClient()
+        
+        test_cases = [
```

Comment:
You can @parametrize pytest tests to make each of these test cases a parameter rather than loop, it's pretty cool

Reply:
Totally agree. I’ll look at how the @parametrize works for these cases. Thanks for the suggestion.